src/providers/node.rs

```diff
@@ -83,9 +120,19 @@ impl Provider for NodeProvider {
 
     fn build(&self, app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
         let mut build_phase = BuildPhase::default();
-
-        if NodeProvider::has_script(app, "build")? {
-            let pkg_manager = NodeProvider::get_package_manager(app);
+        let pkg_manager = NodeProvider::get_package_manager(app);
+
+        if NodeProvider::is_nx_monorepo(app) {
+            let app_name = NodeProvider::get_nx_app_name(app, _env)?.unwrap();
+            build_phase.add_cmd(format!(
+                "{} run build {} --configuration=production",
+                pkg_manager, app_name
+            ));
+            // Remove the package.json so when we run node /dist/**/file.js node does
+            // not look at the package.json and infer things it should not, Such as wether its a es module or not.
+            build_phase.add_cmd("rm /app/package.json".to_owned());
```

Comment:
I am a little confused by this. It is not common to remove the package.json file. Why is it necessary in this case? I've run many NX apps without the need for removing this file. I also know a few apps that rely on this app existing in production.

This also won't work if the start command is overriden to be something like `npm run web:start`

Reply:
The reason I chose not to use the `serve` target is that it appears to be for development purposes as it includes a websocket for use with a  debugger and hot-reload on file change. If this is ok we can totally use `nx run <app-name>:serve:production` see the output below. The other three targets that are generated by the default express generator (`@nrwl/express:aplication`) are `lint`, `build` and `test`.

```
npx nx run express-app:serve:production

> nx run express-app:serve:production

chunk (runtime: main) main.js (main) 564 bytes [entry] [rendered]
webpack compiled successfully (02f31e3162b7080a)
Debugger listening on ws://localhost:9229/666612ae-1046-4d7f-aefd-f39ef9b3fc16
Debugger listening on ws://localhost:9229/666612ae-1046-4d7f-aefd-f39ef9b3fc16
For help, see: https://nodejs.org/en/docs/inspector
nx express app works
chunk (runtime: main) main.js (main) 572 bytes [entry] [rendered]
webpack compiled successfully (4275d6c19665a755)
Command failed: taskkill /pid 28756 /T /F
ERROR: The process "28756" not found.

Debugger listening on ws://localhost:9229/f9432e74-f926-4596-8ad1-a6a5175ef466
Debugger listening on ws://localhost:9229/f9432e74-f926-4596-8ad1-a6a5175ef466
For help, see: https://nodejs.org/en/docs/inspector
nx express app works, reload
```

( same story with `@nrwl/node:webpack` )

The reason for deleting the `package.json` was to solve an issue where running `node <path-to-app>/main.js` would have some interaction with the `package.json`. This iirc. Maybe it would be better ( if we don't use the serve target ) to cd into the output folder and run `node main.js`?

```
ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/app/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```

Reply:
I agree it doesn't make sense to use the `serve` target. But lots of apps have another separate `start` task/script that is meant for production. If that target is found we can run that command. As for the ES module error, I think that can be solved with a field in the package.json file. If the app was deployed and run in other production environments it would not be common to delete the `package.json` file.

Unfortunately deleting this file is not really feasible with Nixpacks since some users may override the start command with the `NIXPACKS_START_CMD` variable to something like `npm run start`. This should work even for NX apps.

Reply:
+1, I agree that start command priority order makes sense.

As for editing package.json, I think it is fairly common to do when running generated code in a production environment. I must admit, I haven't written raw JavaScript in a long time (mainly just TS), but changing the module type in package.json should be okay. The main thing is that we should not be manually deleting these files for the user just to run their code.

Reply:
That's fair, ill get on that tomorrow (hopefully) when I have time, serde should make this easy. Thanks for the insights as well.

Reply:
Just removed the `rm package.json` step and ran the tests ran fine, so I think I've been lying about the module-type issue. 
Ill make the start command change and then I can asses if its really an issue